src/solvers/smt2_incremental/smt2_incremental_decision_procedure.cpp

```diff
@@ -0,0 +1,75 @@
+// Author: Diffblue Ltd.
+
+#include "smt2_incremental_decision_procedure.h"
+
+#include <util/expr.h>
+
+smt2_incremental_decision_proceduret::smt2_incremental_decision_proceduret(
+  std::string solver_command)
+  : solver_command{std::move(solver_command)}, number_of_solver_calls{0}
+{
+}
+
+exprt smt2_incremental_decision_proceduret::handle(const exprt &expr)
+{
+  return expr;
+}
+
+exprt smt2_incremental_decision_proceduret::get(const exprt &expr) const
+{
+  INVARIANT(false, "Unimplemented get of:\n  " + expr.pretty(2, 0));
```

Comment:
`UNIMPLEMENTED` rather than `INVARIANT` please?

Reply:
The `UNIMPLEMENTED` macro is actually under the Legacy annotations section in `invariant.h` with a comment of "should not be used in new code" - https://github.com/diffblue/cbmc/blob/develop/src/util/invariant.h#L519

`UNIMPLEMENTED` also does not allow for a reason to be specified. This would prevent me from including the pretty printing of the expression in the printed output.

Reply:
Yeah, I wrote that :-)  I put it in so that the number of `UNIMPLEMENTED` `TODO` and `UNHANDLED_CASE` asserts decreased with time once we had looked at and converted all of the rogue `asset(0)`s.  I think this is one of the legitimate cases where adding them would be OK.  It certainly would be better style than having direct `INVARIANT(false, "...")`.

Adding a variant of `UNIMPLEMENTED` which takes a message would be fine, as would defining your own custom macro:

```
#define SMT_LIB_WIP(X) INVARIANT(false,"SMT-LIB backend WIP : " X)
```

which would allow you to quickly locate how many of these there are in the code and filter out results that are caused by WIP issues, while running larger scale testing.

Reply:
I would also prefer this to be anything but an `INVARIANT`.

Reply:
This is now using the new `UNIMPLEMENTED_FEATURE` marco.